// 3d-tiles-tools/build/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MeshPrimitiveStructuralMetadata = exports.ElementStructuralMetadata = exports.StructuralMetadataPropertyAttributeProperty = exports.StructuralMetadataPropertyAttribute = exports.StructuralMetadataPropertyTextureProperty = exports.StructuralMetadataPropertyTexture = exports.StructuralMetadataPropertyTableProperty = exports.StructuralMetadataPropertyTable = exports.StructuralMetadataEnumValue = exports.StructuralMetadataEnum = exports.StructuralMetadataClassProperty = exports.StructuralMetadataClass = exports.StructuralMetadataSchema = exports.StructuralMetadata = exports.FeatureIdTexture = exports.MeshFeaturesFeatureId = exports.MeshFeatures = exports.InstanceFeaturesFeatureId = exports.InstanceFeatures = void 0;
__exportStar(require("./base/Buffers"), exports);
__exportStar(require("./base/DataError"), exports);
__exportStar(require("./base/defaultValue"), exports);
__exportStar(require("./base/defined"), exports);
__exportStar(require("./base/DeveloperError"), exports);
__exportStar(require("./base/Iterables"), exports);
__exportStar(require("./base/Paths"), exports);
__exportStar(require("./base/Uris"), exports);
__exportStar(require("./binary/BinaryBufferData"), exports);
__exportStar(require("./binary/BinaryBufferDataResolver"), exports);
__exportStar(require("./binary/BinaryBuffers"), exports);
__exportStar(require("./binary/BinaryBufferStructure"), exports);
__exportStar(require("./binary/BinaryDataError"), exports);
__exportStar(require("./contentProcessing/ContentError"), exports);
__exportStar(require("./contentProcessing/ContentOps"), exports);
__exportStar(require("./contentProcessing/ContentUpgrades"), exports);
__exportStar(require("./contentProcessing/GltfPack"), exports);
__exportStar(require("./contentProcessing/GltfPackOptions"), exports);
__exportStar(require("./contentProcessing/GltfPipelineLegacy"), exports);
__exportStar(require("./contentProcessing/GltfTransform"), exports);
__exportStar(require("./contentProcessing/GltfTransformTextures"), exports);
__exportStar(require("./contentProcessing/GltfUtilities"), exports);
__exportStar(require("./contentProcessing/draco/AttributeInfo"), exports);
__exportStar(require("./contentProcessing/draco/ComponentDataType"), exports);
__exportStar(require("./contentProcessing/draco/DracoDecoder"), exports);
__exportStar(require("./contentProcessing/draco/DracoDecoderResult"), exports);
__exportStar(require("./contentProcessing/draco/DracoError"), exports);
__exportStar(require("./contentProcessing/draco/QuantizationInfo"), exports);
__exportStar(require("./contentProcessing/pointClouds/DefaultPointCloud"), exports);
__exportStar(require("./contentProcessing/pointClouds/GltfTransformPointClouds"), exports);
__exportStar(require("./contentProcessing/pointClouds/PntsPointClouds"), exports);
__exportStar(require("./contentProcessing/pointClouds/ReadablePointCloud"), exports);
__exportStar(require("./contentTypes/BufferedContentData"), exports);
__exportStar(require("./contentTypes/ContentData"), exports);
__exportStar(require("./contentTypes/ContentDataTypeChecks"), exports);
__exportStar(require("./contentTypes/ContentDataTypeEntry"), exports);
__exportStar(require("./contentTypes/ContentDataTypeRegistry"), exports);
__exportStar(require("./contentTypes/ContentDataTypes"), exports);
__exportStar(require("./contentTypes/LazyContentData"), exports);
__exportStar(require("./gltfExtensions/EXTInstanceFeatures"), exports);
__exportStar(require("./gltfExtensions/EXTMeshFeatures"), exports);
__exportStar(require("./gltfExtensions/EXTStructuralMetadata"), exports);
var InstanceFeatures_1 = require("./gltfExtensions/InstanceFeatures");
Object.defineProperty(exports, "InstanceFeatures", { enumerable: true, get: function () { return InstanceFeatures_1.InstanceFeatures; } });
var InstanceFeatures_2 = require("./gltfExtensions/InstanceFeatures");
Object.defineProperty(exports, "InstanceFeaturesFeatureId", { enumerable: true, get: function () { return InstanceFeatures_2.FeatureId; } });
__exportStar(require("./gltfExtensions/InstanceFeaturesUtils"), exports);
var MeshFeatures_1 = require("./gltfExtensions/MeshFeatures");
Object.defineProperty(exports, "MeshFeatures", { enumerable: true, get: function () { return MeshFeatures_1.MeshFeatures; } });
var MeshFeatures_2 = require("./gltfExtensions/MeshFeatures");
Object.defineProperty(exports, "MeshFeaturesFeatureId", { enumerable: true, get: function () { return MeshFeatures_2.FeatureId; } });
var MeshFeatures_3 = require("./gltfExtensions/MeshFeatures");
Object.defineProperty(exports, "FeatureIdTexture", { enumerable: true, get: function () { return MeshFeatures_3.FeatureIdTexture; } });
__exportStar(require("./gltfExtensions/MeshFeaturesUtils"), exports);
var StructuralMetadata_1 = require("./gltfExtensions/StructuralMetadata");
Object.defineProperty(exports, "StructuralMetadata", { enumerable: true, get: function () { return StructuralMetadata_1.StructuralMetadata; } });
var StructuralMetadata_2 = require("./gltfExtensions/StructuralMetadata");
Object.defineProperty(exports, "StructuralMetadataSchema", { enumerable: true, get: function () { return StructuralMetadata_2.Schema; } });
var StructuralMetadata_3 = require("./gltfExtensions/StructuralMetadata");
Object.defineProperty(exports, "StructuralMetadataClass", { enumerable: true, get: function () { return StructuralMetadata_3.Class; } });
var StructuralMetadata_4 = require("./gltfExtensions/StructuralMetadata");
Object.defineProperty(exports, "StructuralMetadataClassProperty", { enumerable: true, get: function () { return StructuralMetadata_4.ClassProperty; } });
var StructuralMetadata_5 = require("./gltfExtensions/StructuralMetadata");
Object.defineProperty(exports, "StructuralMetadataEnum", { enumerable: true, get: function () { return StructuralMetadata_5.Enum; } });
var StructuralMetadata_6 = require("./gltfExtensions/StructuralMetadata");
Object.defineProperty(exports, "StructuralMetadataEnumValue", { enumerable: true, get: function () { return StructuralMetadata_6.EnumValue; } });
var StructuralMetadata_7 = require("./gltfExtensions/StructuralMetadata");
Object.defineProperty(exports, "StructuralMetadataPropertyTable", { enumerable: true, get: function () { return StructuralMetadata_7.PropertyTable; } });
var StructuralMetadata_8 = require("./gltfExtensions/StructuralMetadata");
Object.defineProperty(exports, "StructuralMetadataPropertyTableProperty", { enumerable: true, get: function () { return StructuralMetadata_8.PropertyTableProperty; } });
var StructuralMetadata_9 = require("./gltfExtensions/StructuralMetadata");
Object.defineProperty(exports, "StructuralMetadataPropertyTexture", { enumerable: true, get: function () { return StructuralMetadata_9.PropertyTexture; } });
var StructuralMetadata_10 = require("./gltfExtensions/StructuralMetadata");
Object.defineProperty(exports, "StructuralMetadataPropertyTextureProperty", { enumerable: true, get: function () { return StructuralMetadata_10.PropertyTextureProperty; } });
var StructuralMetadata_11 = require("./gltfExtensions/StructuralMetadata");
Object.defineProperty(exports, "StructuralMetadataPropertyAttribute", { enumerable: true, get: function () { return StructuralMetadata_11.PropertyAttribute; } });
var StructuralMetadata_12 = require("./gltfExtensions/StructuralMetadata");
Object.defineProperty(exports, "StructuralMetadataPropertyAttributeProperty", { enumerable: true, get: function () { return StructuralMetadata_12.PropertyAttributeProperty; } });
var StructuralMetadata_13 = require("./gltfExtensions/StructuralMetadata");
Object.defineProperty(exports, "ElementStructuralMetadata", { enumerable: true, get: function () { return StructuralMetadata_13.ElementStructuralMetadata; } });
var StructuralMetadata_14 = require("./gltfExtensions/StructuralMetadata");
Object.defineProperty(exports, "MeshPrimitiveStructuralMetadata", { enumerable: true, get: function () { return StructuralMetadata_14.MeshPrimitiveStructuralMetadata; } });
__exportStar(require("./gltfExtensions/StructuralMetadataPropertyTables"), exports);
__exportStar(require("./gltfExtensions/StructuralMetadataUtils"), exports);
__exportStar(require("./gltfExtensions/StringBuilder"), exports);
__exportStar(require("./implicitTiling/AvailabilityInfo"), exports);
__exportStar(require("./implicitTiling/AvailabilityInfos"), exports);
__exportStar(require("./implicitTiling/BinarySubtreeData"), exports);
__exportStar(require("./implicitTiling/BinarySubtreeDataResolver"), exports);
__exportStar(require("./implicitTiling/BufferAvailabilityInfo"), exports);
__exportStar(require("./implicitTiling/ConstantAvailabilityInfo"), exports);
__exportStar(require("./implicitTiling/ImplicitTilingError"), exports);
__exportStar(require("./implicitTiling/ImplicitTilings"), exports);
__exportStar(require("./implicitTiling/SubtreeInfo"), exports);
__exportStar(require("./implicitTiling/SubtreeInfos"), exports);
__exportStar(require("./implicitTiling/TemplateUris"), exports);
__exportStar(require("./io/FileResourceResolver"), exports);
__exportStar(require("./io/ResourceResolver"), exports);
__exportStar(require("./io/ResourceResolvers"), exports);
__exportStar(require("./io/TilesetSourceResourceResolver"), exports);
__exportStar(require("./io/UnzippingResourceResolver"), exports);
__exportStar(require("./ktx/BasisEncoder"), exports);
__exportStar(require("./ktx/KtxError"), exports);
__exportStar(require("./ktx/KtxEtc1sOptions"), exports);
__exportStar(require("./ktx/KtxOptions"), exports);
__exportStar(require("./ktx/KtxUastcOptions"), exports);
__exportStar(require("./ktx/KtxUtility"), exports);
__exportStar(require("./logging/Loggers"), exports);
__exportStar(require("./metadata/ArrayValues"), exports);
__exportStar(require("./metadata/ClassProperties"), exports);
__exportStar(require("./metadata/DefaultMetadataEntityModel"), exports);
__exportStar(require("./metadata/DefaultPropertyModel"), exports);
__exportStar(require("./metadata/DefaultPropertyTableModel"), exports);
__exportStar(require("./metadata/MetadataComponentTypes"), exports);
__exportStar(require("./metadata/MetadataEntityModel"), exports);
__exportStar(require("./metadata/MetadataEntityModels"), exports);
__exportStar(require("./metadata/MetadataError"), exports);
__exportStar(require("./metadata/MetadataTypes"), exports);
__exportStar(require("./metadata/MetadataUtilities"), exports);
__exportStar(require("./metadata/MetadataValues"), exports);
__exportStar(require("./metadata/PropertyModel"), exports);
__exportStar(require("./metadata/PropertyModels"), exports);
__exportStar(require("./metadata/PropertyTableModel"), exports);
__exportStar(require("./metadata/PropertyTableModels"), exports);
__exportStar(require("./metadata/TableMetadataEntityModel"), exports);
__exportStar(require("./metadata/binary/ArrayBuffers"), exports);
__exportStar(require("./metadata/binary/BinaryEnumInfo"), exports);
__exportStar(require("./metadata/binary/BinaryMetadata"), exports);
__exportStar(require("./metadata/binary/BinaryPropertyModels"), exports);
__exportStar(require("./metadata/binary/BinaryPropertyTable"), exports);
__exportStar(require("./metadata/binary/BinaryPropertyTableBuilder"), exports);
__exportStar(require("./metadata/binary/BinaryPropertyTableModel"), exports);
__exportStar(require("./metadata/binary/BinaryPropertyTables"), exports);
__exportStar(require("./metadata/binary/BooleanArrayPropertyModel"), exports);
__exportStar(require("./metadata/binary/BooleanPropertyModel"), exports);
__exportStar(require("./metadata/binary/NumericArrayPropertyModel"), exports);
__exportStar(require("./metadata/binary/NumericBuffers"), exports);
__exportStar(require("./metadata/binary/NumericPropertyModel"), exports);
__exportStar(require("./metadata/binary/StringArrayPropertyModel"), exports);
__exportStar(require("./metadata/binary/StringPropertyModel"), exports);
__exportStar(require("./migration/AccessorCreation"), exports);
__exportStar(require("./migration/BatchTableClassProperties"), exports);
__exportStar(require("./migration/BatchTablePropertyTableModels"), exports);
__exportStar(require("./migration/BatchTableSchemas"), exports);
__exportStar(require("./migration/Ids"), exports);
__exportStar(require("./migration/NumberTypeDescriptions"), exports);
__exportStar(require("./migration/TileFormatsMigration"), exports);
__exportStar(require("./migration/TileFormatsMigrationB3dm"), exports);
__exportStar(require("./migration/TileFormatsMigrationI3dm"), exports);
__exportStar(require("./migration/TileFormatsMigrationPnts"), exports);
__exportStar(require("./migration/TileTableDataToMeshFeatures"), exports);
__exportStar(require("./migration/TileTableDataToStructuralMetadata"), exports);
__exportStar(require("./migration/TypeDetection"), exports);
__exportStar(require("./packages/ArchiveFunctions3tz"), exports);
__exportStar(require("./packages/IndexBuilder"), exports);
__exportStar(require("./packages/IndexBuilderEntry"), exports);
__exportStar(require("./packages/IndexEntry"), exports);
__exportStar(require("./packages/TableStructureValidator"), exports);
__exportStar(require("./packages/TilesetSource3dtiles"), exports);
__exportStar(require("./packages/TilesetSource3tz"), exports);
__exportStar(require("./packages/TilesetTarget3dtiles"), exports);
__exportStar(require("./packages/TilesetTarget3tz"), exports);
__exportStar(require("./packages/ZipToPackage"), exports);
__exportStar(require("./spatial/MortonOrder"), exports);
__exportStar(require("./spatial/OctreeCoordinates"), exports);
__exportStar(require("./spatial/Octrees"), exports);
__exportStar(require("./spatial/QuadtreeCoordinates"), exports);
__exportStar(require("./spatial/Quadtrees"), exports);
__exportStar(require("./spatial/TreeCoordinates"), exports);
__exportStar(require("./structure/Asset"), exports);
__exportStar(require("./structure/Availability"), exports);
__exportStar(require("./structure/BoundingVolume"), exports);
__exportStar(require("./structure/BufferObject"), exports);
__exportStar(require("./structure/BufferView"), exports);
__exportStar(require("./structure/Content"), exports);
__exportStar(require("./structure/Group"), exports);
__exportStar(require("./structure/MetadataEntity"), exports);
__exportStar(require("./structure/Properties"), exports);
__exportStar(require("./structure/PropertyTable"), exports);
__exportStar(require("./structure/PropertyTableProperty"), exports);
__exportStar(require("./structure/RootProperty"), exports);
__exportStar(require("./structure/Statistics"), exports);
__exportStar(require("./structure/StatisticsClass"), exports);
__exportStar(require("./structure/StatisticsClassProperty"), exports);
__exportStar(require("./structure/Subtree"), exports);
__exportStar(require("./structure/Subtrees"), exports);
__exportStar(require("./structure/Tile"), exports);
__exportStar(require("./structure/TileImplicitTiling"), exports);
__exportStar(require("./structure/Tileset"), exports);
__exportStar(require("./structure/extensions/BoundingVolumeS2"), exports);
__exportStar(require("./structure/Metadata/ClassProperty"), exports);
__exportStar(require("./structure/Metadata/EnumValue"), exports);
__exportStar(require("./structure/Metadata/MetadataClass"), exports);
__exportStar(require("./structure/Metadata/MetadataEnum"), exports);
__exportStar(require("./structure/Metadata/Schema"), exports);
__exportStar(require("./structure/Style/Style"), exports);
__exportStar(require("./structure/TileFormats/B3dmFeatureTable"), exports);
__exportStar(require("./structure/TileFormats/BatchTable"), exports);
__exportStar(require("./structure/TileFormats/BatchTableBinaryBodyReference"), exports);
__exportStar(require("./structure/TileFormats/BinaryBodyOffset"), exports);
__exportStar(require("./structure/TileFormats/FeatureTable"), exports);
__exportStar(require("./structure/TileFormats/FeatureTableBinaryBodyReference"), exports);
__exportStar(require("./structure/TileFormats/I3dmFeatureTable"), exports);
__exportStar(require("./structure/TileFormats/PntsFeatureTable"), exports);
__exportStar(require("./tileFormats/CompositeTileData"), exports);
__exportStar(require("./tileFormats/Header"), exports);
__exportStar(require("./tileFormats/Table"), exports);
__exportStar(require("./tileFormats/TileData"), exports);
__exportStar(require("./tileFormats/TileDataLayouts"), exports);
__exportStar(require("./tileFormats/TileFormatError"), exports);
__exportStar(require("./tileFormats/TileFormats"), exports);
__exportStar(require("./tilesetData/TilesetEntry"), exports);
__exportStar(require("./tilesetData/TilesetError"), exports);
__exportStar(require("./tilesetData/TilesetInMemory"), exports);
__exportStar(require("./tilesetData/TilesetSource"), exports);
__exportStar(require("./tilesetData/TilesetSourceFs"), exports);
__exportStar(require("./tilesetData/TilesetSources"), exports);
__exportStar(require("./tilesetData/TilesetTarget"), exports);
__exportStar(require("./tilesetData/TilesetTargetFs"), exports);
__exportStar(require("./tilesetData/TilesetTargets"), exports);
__exportStar(require("./tilesetProcessing/BasicTilesetProcessor"), exports);
__exportStar(require("./tilesetProcessing/BoundingVolumes"), exports);
__exportStar(require("./tilesetProcessing/TileContentProcessing"), exports);
__exportStar(require("./tilesetProcessing/TileContentProcessor"), exports);
__exportStar(require("./tilesetProcessing/TileContentProcessors"), exports);
__exportStar(require("./tilesetProcessing/TileContentProcessorsGltfpack"), exports);
__exportStar(require("./tilesetProcessing/TileContentProcessorsGltfPipeline"), exports);
__exportStar(require("./tilesetProcessing/TileContentProcessorsGltfTransform"), exports);
__exportStar(require("./tilesetProcessing/TileContentProcessorsTextures"), exports);
__exportStar(require("./tilesetProcessing/TilesetCombiner"), exports);
__exportStar(require("./tilesetProcessing/TilesetConverter"), exports);
__exportStar(require("./tilesetProcessing/TilesetDataProcessor"), exports);
__exportStar(require("./tilesetProcessing/TilesetEntryProcessor"), exports);
__exportStar(require("./tilesetProcessing/TilesetJsonCreator"), exports);
__exportStar(require("./tilesetProcessing/TilesetMerger"), exports);
__exportStar(require("./tilesetProcessing/TilesetProcessing"), exports);
__exportStar(require("./tilesetProcessing/TilesetProcessor"), exports);
__exportStar(require("./tilesetProcessing/TilesetProcessorContext"), exports);
__exportStar(require("./tilesetProcessing/TilesetProcessorContexts"), exports);
__exportStar(require("./tilesetProcessing/TilesetUpgrader"), exports);
__exportStar(require("./tilesetProcessing/upgrade/TilesetObjectUpgrader"), exports);
__exportStar(require("./tilesetProcessing/upgrade/TilesetUpgradeOptions"), exports);
__exportStar(require("./tilesets/Contents"), exports);
__exportStar(require("./tilesets/Extensions"), exports);
__exportStar(require("./tilesets/Tiles"), exports);
__exportStar(require("./tilesets/Tilesets"), exports);
__exportStar(require("./tilesets/TileTraversalCallback"), exports);
__exportStar(require("./tileTableData/AttributeCompression"), exports);
__exportStar(require("./tileTableData/BatchTables"), exports);
__exportStar(require("./tileTableData/Colors"), exports);
__exportStar(require("./tileTableData/TileTableData"), exports);
__exportStar(require("./tileTableData/TileTableDataI3dm"), exports);
__exportStar(require("./tileTableData/TileTableDataPnts"), exports);
__exportStar(require("./tileTableData/VecMath"), exports);
__exportStar(require("./traversal/ExplicitTraversedTile"), exports);
__exportStar(require("./traversal/ExplicitTraversedTiles"), exports);
__exportStar(require("./traversal/ImplicitTraversedTile"), exports);
__exportStar(require("./traversal/MetadataSemanticOverrides"), exports);
__exportStar(require("./traversal/SubtreeMetadataModel"), exports);
__exportStar(require("./traversal/SubtreeMetadataModels"), exports);
__exportStar(require("./traversal/SubtreeModel"), exports);
__exportStar(require("./traversal/SubtreeModels"), exports);
__exportStar(require("./traversal/TilesetTraverser"), exports);
__exportStar(require("./traversal/TilesetTraversers"), exports);
__exportStar(require("./traversal/TraversalCallback"), exports);
__exportStar(require("./traversal/TraversedTile"), exports);
